File: LMIS/LMIS.Web/Modules/ServerTypes/InventoryManage/DownBookshelfColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DownBookshelfRow } from "./DownBookshelfRow";

export interface DownBookshelfColumns {
    DownBookshelfId: Column<DownBookshelfRow>;
    BookName: Column<DownBookshelfRow>;
    BookshelfLocation: Column<DownBookshelfRow>;
    Inventory: Column<DownBookshelfRow>;
    OperateUserName: Column<DownBookshelfRow>;
    Remark: Column<DownBookshelfRow>;
    CreateTime: Column<DownBookshelfRow>;
    UpdateTime: Column<DownBookshelfRow>;
}

export class DownBookshelfColumns extends ColumnsBase<DownBookshelfRow> {
    static readonly columnsKey = 'InventoryManage.DownBookshelf';
    static readonly Fields = fieldsProxy<DownBookshelfColumns>();
}

File: LMIS/LMIS.Web/Modules/InventoryManage/DownBookshelf/DownBookshelfColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.InventoryManage.Columns;

[ColumnsScript("InventoryManage.DownBookshelf")]
[BasedOnRow(typeof(DownBookshelfRow), CheckNames = true)]
public class DownBookshelfColumns
{
    [AlignRight]
    public long DownBookshelfId { get; set; }
    public string BookName { get; set; }
    public string BookshelfLocation { get; set; }
    public long Inventory { get; set; }
    public string OperateUserName { get; set; }
    [Width(200)]
    public string Remark { get; set; }

    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime CreateTime { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}